frontend/src/components/Sidebar/SidebarAddClassModal.tsx

```diff
@@ -0,0 +1,257 @@
+      semesterList.push(years[i].summer1);
+      semesterList.push(years[i].summer2);
+    }
+
+    this.setState({
+      semesters: semesterList,
+      formSemester: semesterList[0],
+    });
+  }
+
+  async mapRequirementToSchedule() {
+    this.setState({
+      queuedCourses: [],
+    });
+
+    for (let i in this.props.selectedCourses) {
+      const courseToAdd = await fetchCourse(
+        this.props.selectedCourses[i].subject.toUpperCase(),
+        this.props.selectedCourses[i].classId.toString()
+      );
+
+      if (courseToAdd != null) {
+        let curCourses = this.state.queuedCourses.slice();
+        curCourses.push(courseToAdd);
+        this.setState({
+          queuedCourses: curCourses,
+        });
+      }
+    }
+  }
+
+  render() {
+    const { visible } = this.props;
+
+    return (
+      <Modal
+        style={{ outline: 0 }}
+        open={visible}
+        onClose={this.prepareToClose.bind(this)}
+        aria-labelledby="simple-modal-title"
+        aria-describedby="simple-modal-description"
+      >
+        <InnerSection>
+          <CloseButtonWrapper>
+            <XButton onClick={this.prepareToClose.bind(this)}></XButton>
+          </CloseButtonWrapper>
+          <h1 id="simple-modal-title">Add classes</h1>
+          <form>
+            <FormRow>
+              <StyledLabel>
+                Semester:
+                <select
+                  defaultValue={0}
+                  onChange={this.handleSemesterChange.bind(this)}
+                >
+                  {this.state.semesters.map(function(sem, index) {
+                    let curSeason, curYear;
+                    switch (sem.season) {
+                      case "FL":
+                        curSeason = "Fall";
+                        curYear = sem.year;
+                        break;
+                      case "SP":
+                        curSeason = "Spring";
+                        curYear = sem.year + 1;
+                        break;
+                      case "S1":
+                        curSeason = "Summer 1";
+                        curYear = sem.year + 1;
+                        break;
+                      case "S2":
+                        curSeason = "Summer 2";
+                        curYear = sem.year + 1;
+                        break;
+                      default:
+                        curSeason = sem.season;
+                        curYear = sem.year;
+                    }
+
+                    return (
+                      <option
+                        key={sem.id}
+                        value={index}
+                      >{`${curSeason} ${curYear}`}</option>
+                    );
+                  })}
+                </select>
+              </StyledLabel>
+            </FormRow>
+          </form>
+          <SubmitButton onClick={this.handleSubmit.bind(this)}>
+            Add Class
+          </SubmitButton>
+        </InnerSection>
+      </Modal>
+    );
+  }
+}
+
```

Comment:
you should add brief comments about what these objects do in the context of connect. as an example you can check. `https://github.com/sandboxnu/graduatenu/blob/fetch-majors-plans/frontend/src/Onboarding/MajorScreen.tsx#L166`